### PR TITLE
Refs #25217 - Fix target="_blank" vulnerability

### PR DESCRIPTION
**Pull request by sharvit**
More info about the vulnerability:
https://www.jitbit.com/alexblog/256-targetblank---the-most-underestimated-vulnerability-ever/

### CODE REVIEW

**Comment by ohadlevy** [comment]
can you verify this doesn't exists in other parts ? e.g. https://github.com/theforeman/foreman/search?q=_blank&unscoped_q=_blank

**Comment by sharvit** [comment]
@ohadlevy, I ran over all the _blank I could find and updated accordingly.

File: webpack/assets/javascripts/react_app/redux/actions/notifications/index.js [hunk 1]
```diff
@@ -136,5 +136,9 @@ export const toggleDrawer = () => (dispatch, getState) => {
 
 export const clickedLink = link => (dispatch, getState) => {
   toggleDrawer()(dispatch, getState);
-  window.open(link.href, link.external ? '_blank' : '_self');
+  const newWindow = window.open(link.href, link.external ? '_blank' : '_self');
+
+  if (link.external) {
+    newWindow.opener = null;
```
Comment (ohadlevy) [inline webpack/assets/javascripts/react_app/redux/actions/notifications/index.js]: does it make sense to add test for this?

Comment (sharvit) [reply webpack/assets/javascripts/react_app/redux/actions/notifications/index.js]: Test added. Had to add some tweaks to make it testable.

File: webpack/assets/javascripts/react_app/redux/actions/notifications/index.js [hunk 2]
```diff
@@ -134,7 +135,22 @@ export const toggleDrawer = () => (dispatch, getState) => {
...
+  dispatch(toggleDrawerAction());
+
+  const openedWindow = window.open(href, external ? '_blank' : '_self');
+
+  if (external) {
+    openedWindow.opener = null;
+  }
+
+  dispatch({
+    type: NOTIFICATIONS_LINK_CLICKED,
```
Comment (ohadlevy) [inline webpack/assets/javascripts/react_app/redux/actions/notifications/index.js:151]: what am I missing, where is the reducer part?

Comment (sharvit) [reply webpack/assets/javascripts/react_app/redux/actions/notifications/index.js:151]: There is no reducer for link-clicked (add there is no need to) but I still see the value of dispatching the message.

1. This is what actions should do
2. It helps with testing
3. It helps with debugging
4. For future use

**Review by ohadlevy — APPROVED** [review]

**Pull request merged**